core/src/main/java/org/elasticsearch/bootstrap/JVMCheck.java

```diff
@@ -118,7 +118,7 @@ static void check() {
             HotspotBug bug = JVM_BROKEN_HOTSPOT_VERSIONS.get(Constants.JVM_VERSION);
             if (bug != null) {
                 if (bug.workAround != null && ManagementFactory.getRuntimeMXBean().getInputArguments().contains(bug.workAround)) {
-                    Loggers.getLogger(JVMCheck.class).warn(bug.getWarningMessage());
+                    Loggers.getLogger(JVMCheck.class).warn("{}", bug.getWarningMessage());
```

Comment:
Ah. I get it. You are checking that the first string is always a constant. Sometimes it makes for weird things like this but that should be ok.